test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/LinkedHashMapTest.scala

```diff
@@ -47,11 +47,11 @@ abstract class LinkedHashMapTest extends HashMapTest {
   val withSizeLimit = factory.withSizeLimit
 
   @Test def should_iterate_in_insertion_order_after_building(): Unit = {
-    val lhm = factory.empty[jl.Integer, String]
-    (0 until 100).foreach(key => lhm.put(key, s"elem $key"))
+    val lhm = factory.empty[String, String]
+    (0 until 100).foreach(key => lhm.put(key.toString(), s"elem $key"))
```

Comment:
It is unclear to me from the commit message why we change the keys to Strings.

Reply:
I'll add something to the commit message. The gist is that integers have a hashcode that is completely predictable *and in-order*, so testing with integer keys, *especially* when we add them in order from 0 to n, does not exercise any of the corner cases (in particular, insertion order can be observed most of the time even on a regular `HashMap`).